### PR TITLE
SDI-71: 🐛 Improve transfer query performance by adding agency filter

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/repository/sql/MovementsRepositorySql.kt [hunk 1]
```diff
@@ -514,6 +514,7 @@ enum class MovementsRepositorySql(val sql: String) {
                 LEFT JOIN AGENCY_LOCATIONS AL2   ON AL2.AGY_LOC_ID = OIS.TO_AGY_LOC_ID
                 LEFT JOIN REFERENCE_CODES RC3    ON RC3.CODE = OIS.TO_CITY_CODE AND RC3.DOMAIN = 'CITY'
         WHERE OIS.EVENT_DATE = :date
+        AND (OIS.AGY_LOC_ID IN (:agencyListFrom) OR OIS.TO_AGY_LOC_ID IN (:agencyListTo))
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/sql/MovementsRepositorySql.kt:517]: this was getting all schedules for a particular date - now changed to filter in the oracle by the agency

File: src/main/java/uk/gov/justice/hmpps/prison/service/MovementsService.java [hunk 2]
```diff
@@ -320,9 +317,8 @@ private void logErrorAndThrowBadRequest(final String errorMessage) {
         final var datesToTry = splitDatesIfTheySpanAcrossDifferentDays(fromDateTime, toDateTime);
 
         return datesToTry.stream()
-            .flatMap(date -> movementsRepository.getIndividualSchedules(date).stream())
+            .flatMap(date -> movementsRepository.getIndividualSchedules(agencyIds, date).stream())
             .filter(isTransferAndNotDeleted())
-            .filter(isGoingToOrBeingSentFrom(agencyIds))
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/service/MovementsService.java:325]: database is doing filtering instead

File: src/main/java/uk/gov/justice/hmpps/prison/service/MovementsService.java [hunk 3]
```diff
@@ -320,9 +317,8 @@ private void logErrorAndThrowBadRequest(final String errorMessage) {
         final var datesToTry = splitDatesIfTheySpanAcrossDifferentDays(fromDateTime, toDateTime);
 
         return datesToTry.stream()
-            .flatMap(date -> movementsRepository.getIndividualSchedules(date).stream())
+            .flatMap(date -> movementsRepository.getIndividualSchedules(agencyIds, date).stream())
             .filter(isTransferAndNotDeleted())
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/service/MovementsService.java:321]: isn't this filter an "obvious" one to move to database? not sure what other sort events there are but maybe transfer ones are a small minority?

**Review by andymarke — APPROVED** [review]
LGTM
Was there an integration test for this (which of course should remain unchanged)?

**Pull request merged**